scalameta/inputs/shared/src/main/scala/scala/meta/internal/inputs/package.scala

```diff
@@ -6,23 +6,19 @@ import scala.meta.inputs._
 
 package object inputs {
   implicit class XtensionPositionFormatMessage(pos: Position) {
-    def formatMessage(severity: String, message: String): String = {
-      // TODO: In order to be completely compatible with scalac, we need to support Position.point.
-      // On the other hand, do we really need to? Let's try without it. See #383 for discussion.
-      pos.start.formatMessage(severity, message)
+    def formatMessage(severity: String, message: String): String = pos match {
+      case Position.RangeWithPoint(_, _, point, end) =>
+        point.formatMessage(severity, message)
+      case _ =>
+        pos.start.formatMessage(severity, message)
     }
   }
 
   implicit class XtensionPointFormatMessage(point: Point) {
     def formatMessage(severity: String, message: String): String = {
       if (point != Point.None) {
         val input = point.input
-        val shortContent = input match {
-          case Input.File(path, _) => path.toString
-          case Input.LabeledString(label, _) => label
-          case _ => "<input>"
-        }
-        val header = s"$shortContent:${point.line + 1}: $severity: $message"
+        val header = s"${input.syntax}:${point.line + 1}: $severity: $message"
```

Comment:
I have written that pattern match at least 4-5 times in metaconfig/scalafix/scalafmt 😄  This change will be welcome once we clean up pretty-printing in a future release.